### DISCUSSION TITLE
How to use in the CI tools.

### DISCUSSION THREAD

**Discussion by Narven** [onepub-dev/onepub#226] [Support]
Hi, I find the documentation for the CI section extremely confusing.

I need to access a private package that is on onepub. has been published and is available to install. but on the CI pipeline is not downloading due to permission.

What I'm doing currently before the instalation of packages is:

```
dart pub global activate onepub
onepub import
```

My question are:

* this supose to work when I'm on the free plan?
* does this needs the Pro plan to have the `Restricted CI/CD account`, whataver that means?
* is there something else I need to have there on the command?

PS: already have ONEPUB_TOKEN added to env.
PS, part 2: I'm using Codemagic for the CI.

thanks

**Comment by bsutton** [comment]
Pedro,
I believe that I've resolved the issue.

Please re-run `dart pub global activate onepub`
This should install onepub 3.0.1 which has a fix for the issue.

Let me know if you have any issues.

Thanks for your patience.

Brett

**Comment by bsutton** [comment]
A few other points for clairty.

Yes you need at least the Pro plan to use a CI/CD account.
However you don't need to use a CI/CD account, you can use your own account.

The idea behind a CI/CD account is that you can have a dedicated account
that is only used for CI/CD.
It has restricted access (can run dart pub get but not dart pub publish).

I will review the doco to see if I can make this clearer.

Brett

**Comment by Narven** [comment]
Hi,

thanks it looks like is working through the CI pipeline.

Thank u very much :)


Pedro Luz

Software Engineer
https://pedromsluz.com
https://twitter.com/narven

**Comment by bsutton** [comment]
Good to hear

Can I ask what you think of the onepub pricing?

**Reply by Narven** [reply]
I think is a good pricing, has nice balance between features/price.

**Comment by bsutton** [comment]
Appreciate the feed back

**Reply by Narven** [reply]
Let me know if you need any help
